### PR TITLE
fix(charts): fix default dependency service

### LINKED ISSUE #786: TRG 5.01 Helm chart must be deployable to any environment without overwriting default values with a simple helm install command
After another qg check found some helm template for your postgress init step maybe this helps to generate a password if nothing is provided into your product values.yaml, because while installing helm chart without any values there would be a little proplem with the PostgreSql password:

`Caused by: org.flywaydb.core.internal.exception.FlywaySqlException: Unable to obtain connection from database: FATAL: password authentication failed for user "bpdm"------------------------------------------------------------------------------------------------SQL State  : 28P01Error Code : 0Message    : FATAL: password authentication failed for user "bpdm"`

If this is on purpose, please ignore this info.

https://github.com/eclipse-tractusx/sldt-bpn-discovery/blob/f46c54f9606bfc86cf6cacbf0a57ac55c6d5952c/charts/bpndiscovery/values.yaml#L88-L102

https://github.com/eclipse-tractusx/sldt-bpn-discovery/blob/main/charts/bpndiscovery/templates/postgres-init.yaml

### PR DESCRIPTION
**Pull request by nicoprow**
## Description

This pull request fixes how the BPDM Charts resolve default service names of dependencies. The umbrella and subcharts now find the postgres dependency and the password on default without any configuration needed.

Also fixed how the Charts resolve the default service names for other BPDM charts that they are dependend on.

Fixes #786 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by nicoprow** [comment]
Contributes towards #795

**Review by FaGru3n — CHANGES_REQUESTED** [review]
Hi @nicoprow thanks for your fast work.

saw that you removed the fullnameOverwrite setting and i guess with "nameOverwrite" setting should also be fine and guess would help evelyn with theirs issue.

and as stated in TRG 5.01 

> Dependencies should be prefixed with the nameOverride and/or fullnameOverride properties

rest LGTM

File: charts/bpdm/values.yaml [hunk 1]
```diff
@@ -21,20 +21,17 @@
 bpdm-gate:
   enabled: true
   postgres:
-    # use the name that was override on postgres.fullnameOverride to be used by the application on connection
-    fullnameOverride: ""
+    enabled: false
```
[the remark this answers is not in the document]
Comment (nicoprow) [reply charts/bpdm/values.yaml:24]: I don't understand why dependencies necessarily need to be introduced with a nameOverride or fullNameOverride. The postgres dependency name can be overriden in this pull request but it doesn't need to. If not overriden the postges service name will include the release name.

I performed deployment tests for this pull requests chart: One without any configuration changes and one in which I replicated the configuration from #795. Both deploy without any issues and the created services have the overriden names.

[the remark this answers is not in the document]
Comment (FaGru3n) [reply charts/bpdm/values.yaml:24]: regarding to suggested changes, there is a reason for it and guess we should not mix release changes with modification for umbrella helm chart so will approve for now and furhter changes has to be disscussed between the parties itself

**Review by FaGru3n — APPROVED** [review]
LGTM

**Pull request merged**